wpi_interface/src/main/java/coppercore/wpi_interface/MonitoredSubsystem.java

```diff
@@ -0,0 +1,47 @@
+package coppercore.wpi_interface;
+
+import coppercore.monitors.Monitor;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import java.util.ArrayList;
+import java.util.List;
+import org.littletonrobotics.junction.Logger;
+
+public class MonitoredSubsystem extends SubsystemBase {
+    private List<Monitor> registeredMonitors = new ArrayList<Monitor>();
+
+    public void addMonitor(Monitor monitor) {
+        registeredMonitors.add(monitor);
+    }
+
+    @Override
+    public void periodic() {
+        monitoredPeriodic();
+        runMonitors();
+    }
+
+    /**
+     * OVERRIDE ME! This function is called every time the subsystem's periodic function is called.
+     * However, MonitoredSubsytem automatically checks monitors during every periodic run.
+     * Therefore, this method should be overridden as a replacement for the normal periodic function
+     * in the implementation of the subsystem.
+     *
+     * <p>This method is called periodically by the {@link CommandScheduler}. Useful for updating
+     * subsystem-specific state that you don't want to offload to a {@link Command}. Teams should
+     * try to be consistent within their own codebases about which responsibilities will be handled
+     * by Commands, and which will be handled here.
+     */
+    public void monitoredPeriodic() {}
+
+    private void runMonitors() {
+        registeredMonitors.forEach(
+                monitor -> {
+                    monitor.periodic(Timer.getFPGATimestamp());
+
+                    Logger.recordOutput(
+                            "monitors/" + monitor.getName() + "/triggered", monitor.isTriggered());
+                    Logger.recordOutput(
+                            "monitors/" + monitor.getName() + "/faulted", monitor.isFaulted());
```

Comment:
I think it would be good to be able to turn this logging on or off. We run out of memory with lots of strings and we will probably have a bunch of monitors this season.

Maybe the move is to make each monitor subsystem have a flag for if it should be logging these. I'm fine if this gets deferred to a different ticket, but it's just something to think about

Reply:
I've added the ability to enable/disable logging for the MonitoredSubsystem and to enable/disable logging for each individual monitor as well.